src/assets.rs

```diff
@@ -226,33 +222,71 @@ impl<'a> DerefMut for AssetEditor<'a> {
 
 /// Assets stored in an assets store.
 pub struct Assets {
+    /// A key-value store for assets.
     pub store: AssetsStore,
+
+    /// A set of asset codes loaded from verified asset libraries.
+    pub verified_assets: HashSet<AssetCode>,
 }
 
 impl Assets {
     #![allow(dead_code)]
 
-    /// Create assets.
+    /// Create unverified assets.
+    ///
+    /// To load verified assets afterwards, use `verify_asset`.
     pub fn new<L: Ledger>(store: AssetsStore) -> Result<Self, KeystoreError<L>> {
-        Ok(Self { store })
+        Ok(Self {
+            store,
+            verified_assets: HashSet::new(),
+        })
     }
 
     /// Iterate through the assets.
     pub fn iter(&self) -> impl Iterator<Item = Asset> + '_ {
         self.store.iter().cloned()
     }
 
-    /// Get the asset by the code.
+    /// Load a verified asset library with its trusted signer.
+    ///
+    /// Call this function at least once after assets resource is created.
+    pub async fn verify_assets<L: Ledger>(
+        &mut self,
+        trusted_signer: &VerKey,
+        library: VerifiedAssetLibrary,
+    ) -> Result<(), KeystoreError<L>> {
+        if let Some(assets) = library.open(trusted_signer) {
+            for asset in &assets {
+                self.verified_assets.insert(asset.definition.code);
+            }
+            Ok(())
+        } else {
+            Err(KeystoreError::AssetVerificationError)
+        }
+    }
+
+    /// Get the asset by the code from the store.
+    ///
+    /// Sets the asset as verified if it's in the verified set.
     pub fn get<L: Ledger>(&self, code: &AssetCode) -> Result<Asset, KeystoreError<L>> {
-        Ok(self.store.load(code)?)
+        let mut asset = self.store.load(code)?;
+        if self.verified_assets.contains(code) {
+            asset = asset.export_verified();
```

Comment:
Should be `asset.verified = true`. Here we _do not_ want to `export_verified` and clear the mint info, since we are loading an asset that is already in our library, not necessarily for the purposes of exporting, and we need access to the mint info if we want to mint this asset. The caller can choose to call `export_verified` if they do want to export the asset.